extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -2236,7 +2253,8 @@ private static Pair<List<DimensionSchema>, List<AggregatorFactory>> makeDimensio
                 outputColumnAggregatorFactories,
                 outputColumnName,
                 type,
-                query.context()
+                query.context(),
+                dimensionToSchemaMap
```

Comment:
Where are they getting piped to the segment generator factory ?

Reply:
The schema returned from this function call is used as the input to the factory.